From b0dbabc6c63698586cae1a626789b5108bb7de83 Mon Sep 17 00:00:00 2001
From: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
Date: Fri, 22 Jul 2022 16:24:36 +0800
Subject: [PATCH 40/51] GN: Wayland build overrides

Introduce //build_overrides/wayland.gni to simplify the management of
the wayland dependencies in Chromium, ANGLE, and SwiftShader.

Cherry-picked from: commit 3d415c7d0eba78f0b18e2c1fe1e84e4afa7e66e2

Signed-off-by: Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com>
---
 build_overrides/wayland.gni | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)
 create mode 100644 build_overrides/wayland.gni

diff --git a/build_overrides/wayland.gni b/build_overrides/wayland.gni
new file mode 100644
index 000000000000..0d48c7e31a2e
--- /dev/null
+++ b/build_overrides/wayland.gni
@@ -0,0 +1,17 @@
+# Copyright 2022 The Chromium Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("//build/config/ozone.gni")
+
+wayland_gn_dir = "//third_party/wayland"
+
+if (ozone_platform_wayland) {
+  # This defines use_system_libwayland
+  import("$wayland_gn_dir/features.gni")
+} else {
+  use_system_libwayland = false
+}
+
+# Path to Wayland sources
+wayland_dir = "$wayland_gn_dir/src"
-- 
2.25.1

